test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1086,16 +1095,18 @@ private synchronized void reset(boolean wipeData) throws IOException {
 
     /** ensure a cluster is formed with all published nodes. */
     public synchronized void validateClusterFormed() {
-        String name = randomFrom(random, getNodeNames());
+        String name = randomFrom(random, nodes.values().stream()
+            .filter(nc -> nc.isClosed() == false).map(nc -> nc.name).toArray(String[]::new));
         validateClusterFormed(name);
     }
 
     /** ensure a cluster is formed with all published nodes, but do so by using the client of the specified node */
     private synchronized void validateClusterFormed(String viaNode) {
-        Set<DiscoveryNode> expectedNodes = new HashSet<>();
-        for (NodeAndClient nodeAndClient : nodes.values()) {
-            expectedNodes.add(getInstanceFromNode(ClusterService.class, nodeAndClient.node()).localNode());
-        }
+        Set<DiscoveryNode> expectedNodes =
+            nodes.values().stream()
```

Comment:
should we temporarily remove the node from the `nodes` map when it is restarted during the close? This could possibly simplify things here, not requiring us to exclude closed nodes everywhere. WDYT?

Reply:
I originally did not do so due to test failures. In hindsight, I think I agree, it will leave this in a much cleaner state. Will give it a go and see if I can fix the tests that needs fixing (at least GatewayIndexStateIT.testArchiveBrokenClusterSettings.